src/main/java/net/glowstone/GlowWorld.java

```diff
@@ -1208,12 +1208,16 @@ public GlowEntity spawn(Location location, Class<? extends GlowEntity> clazz, Sp
             try {
                 Constructor<? extends GlowEntity> constructor = clazz.getConstructor(Location.class);
                 entity = constructor.newInstance(location);
-                CreatureSpawnEvent spawnEvent = new CreatureSpawnEvent((LivingEntity) entity, reason);
-                if (!spawnEvent.isCancelled()) {
-                    entity.createSpawnMessage();
+                if (entity instanceof LivingEntity) {
+                    CreatureSpawnEvent spawnEvent = new CreatureSpawnEvent((LivingEntity) entity, reason);
+                    if (!spawnEvent.isCancelled()) {
+                        entity.createSpawnMessage();
```

Comment:
I know this was here before, but I'm pretty sure that the spawn message needs to be sent to players that can see the chunk the entity is in.